src/Extensions/Diagnostics.HealthChecks/Internal/RestHealthCheckPublisher.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Omex.Extensions.Abstractions.ExecutionContext;
+using Microsoft.ServiceFabric.Client;
+using Microsoft.ServiceFabric.Client.Http;
+using sfc = Microsoft.ServiceFabric.Common;
+using sfh = System.Fabric.Health;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
+{
+	/// <summary>
+	/// Just mock
+	/// </summary>
+	public class RestHealthCheckPublisher : HealthCheckPublisher
+	{
+		private readonly ServiceFabricHttpClient m_client;
+		private readonly string m_serviceId;
+		/// <summary>
+		/// 
+		/// </summary>
+		protected override string HealthReportSourceIdImpl => nameof(RestHealthCheckPublisher);
+
+		/// <summary>
+		/// 
+		/// </summary>
+		public RestHealthCheckPublisher(Uri clusterEndpoints, string serviceId) : base()
+		{
+			m_serviceId = serviceId;
+			m_client = (ServiceFabricHttpClient)new ServiceFabricClientBuilder()
+				.UseEndpoints(clusterEndpoints)
+				.BuildAsync().GetAwaiter().GetResult();
```

Comment:
Calling `.GetAwaiter().GetResult()` considered a bad practice since it could lead to wrong handling of SyncronizationContext and deadlocks.
I understand that you are unable to make constructor async, so probably the only option here would be to build a client during the first call of `PublishAsync` or consider alternative design approaches that I'll describe in the general comment.

Reply:
Added RestClientWrapper